pkg/symbol/elfutils/elfutils.go

```diff
@@ -153,3 +158,109 @@ func HasSymbols(filePath string) (bool, error) {
 	}
 	return false, nil
 }
+
+// ValidateFile returns an error if the given object file is not valid.
+func ValidateFile(path string) error {
+	elfFile, err := elf.Open(path)
+	if err != nil {
+		return err
+	}
+	defer elfFile.Close()
+
+	// TODO(kakkoyun): How can we improve this without allocating too much memory.
+	if len(elfFile.Sections) == 0 {
+		return errors.New("ELF does not have any sections")
+	}
+	return nil
+}
+
+// ValidateHeader returns an error if the given object file header is not valid.
+func ValidateHeader(r io.Reader) error {
```

Comment:
This is fantastic!! Love how you are validating a bunch of things I did not even think about!! 💯